Modules/TPC/src/CheckOfSlices.cxx

```diff
@@ -110,184 +119,438 @@ Quality CheckOfSlices::check(std::map<std::string, std::shared_ptr<MonitorObject
+    ILOG(Fatal, Support) << "No Graph object found" << ENDM;
   }
-  const int NBins = h->GetN();
-  if (NBins == 0) {
+
+  const int nPoints = h->GetN();
+  if (nPoints == 0) {
     ILOG(Fatal, Support) << "No bins were found for the Graph!" << ENDM;
   }
+
   const double* yValues = h->GetY();
-  const double* yErrors = h->GetEY();
-  const std::vector<double> v(yValues, yValues + NBins); // use all points
-  const std::vector<double> vErr(yErrors, yErrors + NBins);
-  const double sum = std::accumulate(v.begin(), v.end(), 0.0);
-  const double meanFull = sum / NBins;
-  TPaveText* msg = new TPaveText(0.7, 0.85, 0.9, 0.9, "NDC");
-  TPaveText* Legend = new TPaveText(0.1, 0.85, 0.35, 0.9, "NDC");
+  const double* yErrors = h->GetEY(); // returns nullptr for TGraph (no errors)
+  bool useErrors = true;
+  if (yErrors == nullptr) {
+    useErrors = false;
+  }
+
+  TPaveText* msg = new TPaveText(0.5, 0.75, 0.9, 0.9, "NDC");
   h->GetListOfFunctions()->Add(msg);
-  h->GetListOfFunctions()->Add(Legend);
   msg->SetName(fmt::format("{}_msg", mo->GetName()).data());
 
+  std::string checkMessage;
   if (checkResult == Quality::Good) {
-    h->SetFillColor(kGreen);
+    h->SetFillColor(kGreen - 2);
     msg->Clear();
     msg->AddText("Quality::Good");
-    msg->SetFillColor(kGreen);
+    msg->SetFillColor(kGreen - 2);
   } else if (checkResult == Quality::Bad) {
-    ILOG(Info, Support) << "Quality::Bad, setting to red";
     h->SetFillColor(kRed);
     msg->Clear();
-    msg->AddText("Quality::Bad");
-    if (mCheckChoice == CheckChoiceExpectedPhysicsValue) {
-      msg->AddText(fmt::format("Outlier, more than {} sigma.", mNSigmaBadExpectedPhysicsValue).data());
-    } else if (mCheckChoice == CheckChoiceMean) {
-      msg->AddText(fmt::format("Outlier, more than {} sigma.", mNSigmaBadMean).data());
-    } else {
-      msg->AddText("Outlier. Bad Quality.");
-    }
+    msg->AddText("Quality::Bad. Failed checks:");
+    checkMessage = mBadString;
     msg->SetFillColor(kRed);
   } else if (checkResult == Quality::Medium) {
-    ILOG(Info, Support) << "Quality::medium, setting to orange";
     h->SetFillColor(kOrange);
     msg->Clear();
-    msg->AddText("Quality::Medium");
-    if (mCheckChoice == CheckChoiceExpectedPhysicsValue) {
-      msg->AddText(fmt::format("Outlier, more than {} sigma.", mNSigmaExpectedPhysicsValue).data());
-    } else if (mCheckChoice == CheckChoiceMean) {
-      msg->AddText(fmt::format("Outlier, more than {} sigma.", mNSigmaMean).data());
-    } else {
-      msg->AddText("Outlier. Medium Quality");
-    }
+    msg->AddText("Quality::Medium. Failed checks:");
+    checkMessage = mMediumString;
     msg->SetFillColor(kOrange);
   } else if (checkResult == Quality::Null) {
     h->SetFillColor(0);
+    msg->AddText("Quality::Null. Failed checks:");
+    checkMessage = mNullString;
+  }
+
+  // Split lines by hand as \n does not work with TPaveText
+  std::string delimiter = "\n";
```

Comment:
```suggestion
  const std::string delimiter = "\n";
```

Reply:
`Done, I also did a similar change now in CheckOfTrendings`